### DISCUSSION TITLE
ormar /postgres datetime/timezone

### DISCUSSION THREAD

**Discussion by avico78** [collerek/ormar#190] [Q&A] [answered]
How should define field so it will use localtimezone?
Just encoter to situation where Grafana see localtime zone and when field define without timezone 
it cause a delay .

I'm trying to implement something like:

```python

from datetime import timezone
import datetime

class DateFieldsModel(ormar.Model):
    class Meta:
        abstract = True

    created_date: datetime.datetime = ormar.DateTime(
        default=datetime.datetime.now(timezone.utc+3), name="creation_date"
    )
    updated_date: datetime.datetime = ormar.DateTime(
        default=datetime.datetime.now(timezone.utc+3), name="modification_date"
    )

````

getting:
```

File "asyncpg/protocol/protocol.pyx", line 181, in bind_execute

File "asyncpg/protocol/prepared_stmt.pyx", line 171, in asyncpg.protocol.protocol.PreparedStatementState._encode_bind_msg

asyncpg.exceptions.DataError: invalid input for query argument $1: datetime.datetime(2021, 5, 5, 14, 13, 28... (can't subtract offset-naive and offset-aware datetimes)
```

**Comment by collerek** [comment]
When I run the datetime as you wrote it I get an exception.

This works for me:
```python
from datetime import datetime, timezone, timedelta

check = datetime.now(tz=timezone(timedelta(hours=3)))
print(check)
```

But the error is coming from a statement, which statement? It seams you try to filter by date that has no timezone or something similar?

By the way the de facto standard is to keep all the dates either in system timezone (your code server tz) or your database tz (sql server). Most of the systems keep also dates in utc and only convert it to different tzs for presentation, as operations on dates with tzs can be very tricky. If you want system date use `datetime.now` if you want sql date you need to pass `server_default` and set it to `sqlalchemy` function `func.now()`

Note that you also execute now() so you assign a value, one, same date for all, set at start of your script (use lambda or helper to get a new one each time).

**Comment by zdytch** [comment]
Have similar problem

The model field:
`updated_at: datetime = ormar.DateTime(default=lambda: datetime.now(pytz.utc))`

On creating an instance, get the exception:
`asyncpg.exceptions.DataError: invalid input for query argument $4: datetime.datetime(2021, 7, 4, 16, 52, 4,... (can't subtract offset-naive and offset-aware datetimes)`

Have googled an issue: https://github.com/MagicStack/asyncpg/issues/138
Maybe the same reason?

**Reply by zdytch** [reply]
In SQAlchemy we have  the`timezone` parameter for `DateTime` field: https://docs.sqlalchemy.org/en/14/core/type_basics.html#sqlalchemy.types.DateTime

Maybe it makes sense to include it to the Ormar's DateTime model?

**Reply by collerek** [reply]
@zdytch from sqlalchemy docs it looks like a boolean, only enabling the timezone awareness? 

I will look into this, thanks for the suggestion! :)

**Reply by zdytch** [reply]
@collerek thank you!
If I understand right, under the hood it will set Postgres field type from TIMESTAMP to TIMESTAMPTZ

**Comment by collerek** [comment, marked as answer]
Added in 0.10.14 (via #265).
Please update and check.

**Reply by zdytch** [reply]
It works now, great!
